File: docs/specs/esm/bias.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec(
  vg.loadParquet("walk", "data/random-walk.parquet")
);

const $point = vg.Param.value(0);

export default vg.vconcat(
  vg.slider({ label: "Bias", as: $point, min: 1, max: 1000, step: 0.1 }),
  vg.plot(
    vg.areaY(
      vg.from("walk"),
      { x: "t", y: vg.sql`v + ${$point}`, fill: "steelblue" }
    ),
    vg.width(680),
    vg.height(200)
  )
);

File: docs/public/specs/python/bias.py
import json
import vgplot as vg

meta = vg.meta(title="Bias Parameter", description="Dynamically adjust queried values by adding a Param value. The SQL expression is re-computed in the database upon updates.\n")
data = vg.data(
    walk=vg.parquet("data/random-walk.parquet")
)

view = vg.vconcat(
    vg.slider(label="Bias", as_="$point", min=0, max=1000, step=1),
    vg.plot(
            vg.area_y(data=vg.from_("walk"), x="t", y={
                "sql": "v + $point"
            }, fill="steelblue"),
            vg.width(680),
            vg.height(200)
        )
)

params = {
    "point": 0
}

spec = vg.spec(meta=meta, data=data, params=params, view=view)

if __name__ == "__main__":
    print(json.dumps(spec.to_dict(), sort_keys=True))